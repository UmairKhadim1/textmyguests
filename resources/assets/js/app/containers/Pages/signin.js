// @flow
import React, { Component } from 'react';
import store from '../../redux/store';
import { Link } from 'react-router-dom';
import { Form, Input } from 'antd';
import { connect } from 'react-redux';
import {
  AuthStyledWrapper,
  AuthContent,
  FormWrapper,
  StyledButton,
} from './auth.style';

type FormProps = {
  form: Object,
  loading: boolean,
  onLogin: (string, string) => void,
};

class LoginBasic extends Component<FormProps> {
  static defaultProps = {
    onLogin: () => null,
  };

  handleSubmit = e => {
    e.preventDefault();
    this.props.form.validateFieldsAndScroll((err, values) => {
      if (!err) {
        const { email, password } = values;
        this.props.onLogin(email, password);
      }
    });
  };

  render() {
    const { loading, form } = this.props;
    const { getFieldDecorator } = form;
    return (
      <Form onSubmit={this.handleSubmit} className="isoSignInForm">
        <Form.Item className="isoInputWrapper">
          {getFieldDecorator('email', {
            rules: [
              {
                // eslint-disable-next-line no-useless-escape
                pattern: /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
                message: 'Please enter a valid email address',
                required: true,
              },
            ],
          })(<Input placeholder="Email" />)}
        </Form.Item>
        <Form.Item className="isoInputWrapper">
          {getFieldDecorator('password', {
            rules: [
              {
                required: true,
                message: 'This field is required',
              },
            ],
          })(<Input type="password" placeholder="Password" />)}
        </Form.Item>
        <Form.Item>
          <StyledButton
            disabled={loading}
            htmlType="submit"
            loading={loading}
            type="primary">
            Sign In
          </StyledButton>
        </Form.Item>
      </Form>
    );
  }
}

export const LoginForm = Form.create()(LoginBasic);

type Props = {
  login: ({ email: string, password: string, cb: () => void }) => void,
  history: Object,
  loading: boolean,
  inviteToken?: string,
};

class SignIn extends Component<Props> {
  static defaultProps = {
    loading: false,
    isLoggedIn: false,
  };

  handleLogin = (email: string, password: string) => {
    const { login, history } = this.props;
    login({
      email,
      password,
      cb: () => history.push('dashboard'),
    });
  };

  render() {
    return (
      <AuthStyledWrapper className="sign-in">
        <AuthContent>
          <div className="tmg-logo-container">
            <img
              className="white-logo"
              src="/assets/images/live/TMG-Logo-Small.png"
              alt="TextMyGuests logo"
            />
          </div>

          <h1 className="title">Welcome back! Sign in here.</h1>

          <div className="isoHelperWrapper">
            <Link to="/forgot-password" className="isoForgotPass">
              Forgot Password?
            </Link>
            <br />
            <Link to="/signup">Create Account</Link>
          </div>

          <FormWrapper>
            {this.props.inviteToken && (
              <p
                style={{
                  marginBottom: '1rem',
                  color: '#1890ff',
                  fontSize: '15px',
                  fontWeight: 600,
                  textAlign: 'center',
                }}>
                To accept the invitation, simply sign in or create an account.
              </p>
            )}
            <LoginForm
              onLogin={this.handleLogin}
              loading={this.props.loading}
            />
          </FormWrapper>
        </AuthContent>
      </AuthStyledWrapper>
    );
  }
}

export default connect(
  state => ({
    loading: state.loading.effects.Auth.login,
    inviteToken: store.select.Auth.inviteToken(state),
  }),
  ({ Auth }) => ({ login: Auth.login })
)(SignIn);
