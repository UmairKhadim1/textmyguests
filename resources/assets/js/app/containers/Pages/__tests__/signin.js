import React from 'react';
import { LoginForm } from '../signin';
import { shallow } from 'enzyme';
jest.mock('../../../redux/auth/reducer');

describe('Login Form tests', () => {
  it('should render without crash', () => {
    const wrapper = shallow(<LoginForm />).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
