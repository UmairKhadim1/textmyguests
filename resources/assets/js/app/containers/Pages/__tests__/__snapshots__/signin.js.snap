// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form tests should render without crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginBasic
    form={
      Object {
        "getFieldDecorator": [Function],
        "getFieldError": [Function],
        "getFieldInstance": [Function],
        "getFieldProps": [Function],
        "getFieldValue": [Function],
        "getFieldsError": [Function],
        "getFieldsValue": [Function],
        "isFieldTouched": [Function],
        "isFieldValidating": [Function],
        "isFieldsTouched": [Function],
        "isFieldsValidating": [Function],
        "isSubmitting": [Function],
        "resetFields": [Function],
        "setFields": [Function],
        "setFieldsInitialValue": [Function],
        "setFieldsValue": [Function],
        "submit": [Function],
        "validateFields": [Function],
        "validateFieldsAndScroll": [Function],
      }
    }
    onLogin={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <FormItem
          className="isoInputWrapper"
          colon={true}
          hasFeedback={false}
          prefixCls="ant-form"
        >
          <Input
            data-__field={
              Object {
                "name": "email",
              }
            }
            data-__meta={
              Object {
                "name": "email",
                "originalProps": Object {
                  "disabled": false,
                  "placeholder": "Email",
                  "prefixCls": "ant-input",
                  "type": "text",
                },
                "ref": null,
                "rules": Array [
                  Object {
                    "message": "Please enter a valid email address",
                    "pattern": /\\^\\[a-z0-9\\._%\\+-\\]\\+@\\[a-z0-9\\.-\\]\\+\\\\\\.\\[a-z\\]\\{2,\\}\\$/,
                    "required": true,
                  },
                ],
                "trigger": "onChange",
                "validate": Array [
                  Object {
                    "rules": Array [
                      Object {
                        "message": "Please enter a valid email address",
                        "pattern": /\\^\\[a-z0-9\\._%\\+-\\]\\+@\\[a-z0-9\\.-\\]\\+\\\\\\.\\[a-z\\]\\{2,\\}\\$/,
                        "required": true,
                      },
                    ],
                    "trigger": Array [
                      "onChange",
                    ],
                  },
                ],
                "valuePropName": "value",
              }
            }
            disabled={false}
            id="email"
            onChange={[Function]}
            placeholder="Email"
            prefixCls="ant-input"
            type="text"
          />
        </FormItem>,
        <FormItem
          className="isoInputWrapper"
          colon={true}
          hasFeedback={false}
          prefixCls="ant-form"
        >
          <Input
            data-__field={
              Object {
                "name": "password",
              }
            }
            data-__meta={
              Object {
                "name": "password",
                "originalProps": Object {
                  "disabled": false,
                  "placeholder": "Password",
                  "prefixCls": "ant-input",
                  "type": "password",
                },
                "ref": null,
                "rules": Array [
                  Object {
                    "message": "This field is required",
                    "required": true,
                  },
                ],
                "trigger": "onChange",
                "validate": Array [
                  Object {
                    "rules": Array [
                      Object {
                        "message": "This field is required",
                        "required": true,
                      },
                    ],
                    "trigger": Array [
                      "onChange",
                    ],
                  },
                ],
                "valuePropName": "value",
              }
            }
            disabled={false}
            id="password"
            onChange={[Function]}
            placeholder="Password"
            prefixCls="ant-input"
            type="password"
          />
        </FormItem>,
        <FormItem
          colon={true}
          hasFeedback={false}
          prefixCls="ant-form"
        >
          <Button
            block={false}
            ghost={false}
            htmlType="submit"
            loading={false}
            prefixCls="ant-btn"
            type="primary"
          >
            Sign In
          </Button>
        </FormItem>,
      ],
      "className": "isoSignInForm",
      "hideRequiredMark": false,
      "layout": "horizontal",
      "onSubmit": [Function],
      "prefixCls": "ant-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Input
            data-__field={
              Object {
                "name": "email",
              }
            }
            data-__meta={
              Object {
                "name": "email",
                "originalProps": Object {
                  "disabled": false,
                  "placeholder": "Email",
                  "prefixCls": "ant-input",
                  "type": "text",
                },
                "ref": null,
                "rules": Array [
                  Object {
                    "message": "Please enter a valid email address",
                    "pattern": /\\^\\[a-z0-9\\._%\\+-\\]\\+@\\[a-z0-9\\.-\\]\\+\\\\\\.\\[a-z\\]\\{2,\\}\\$/,
                    "required": true,
                  },
                ],
                "trigger": "onChange",
                "validate": Array [
                  Object {
                    "rules": Array [
                      Object {
                        "message": "Please enter a valid email address",
                        "pattern": /\\^\\[a-z0-9\\._%\\+-\\]\\+@\\[a-z0-9\\.-\\]\\+\\\\\\.\\[a-z\\]\\{2,\\}\\$/,
                        "required": true,
                      },
                    ],
                    "trigger": Array [
                      "onChange",
                    ],
                  },
                ],
                "valuePropName": "value",
              }
            }
            disabled={false}
            id="email"
            onChange={[Function]}
            placeholder="Email"
            prefixCls="ant-input"
            type="text"
          />,
          "className": "isoInputWrapper",
          "colon": true,
          "hasFeedback": false,
          "prefixCls": "ant-form",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data-__field": Object {
              "name": "email",
            },
            "data-__meta": Object {
              "name": "email",
              "originalProps": Object {
                "disabled": false,
                "placeholder": "Email",
                "prefixCls": "ant-input",
                "type": "text",
              },
              "ref": null,
              "rules": Array [
                Object {
                  "message": "Please enter a valid email address",
                  "pattern": /\\^\\[a-z0-9\\._%\\+-\\]\\+@\\[a-z0-9\\.-\\]\\+\\\\\\.\\[a-z\\]\\{2,\\}\\$/,
                  "required": true,
                },
              ],
              "trigger": "onChange",
              "validate": Array [
                Object {
                  "rules": Array [
                    Object {
                      "message": "Please enter a valid email address",
                      "pattern": /\\^\\[a-z0-9\\._%\\+-\\]\\+@\\[a-z0-9\\.-\\]\\+\\\\\\.\\[a-z\\]\\{2,\\}\\$/,
                      "required": true,
                    },
                  ],
                  "trigger": Array [
                    "onChange",
                  ],
                },
              ],
              "valuePropName": "value",
            },
            "disabled": false,
            "id": "email",
            "onChange": [Function],
            "placeholder": "Email",
            "prefixCls": "ant-input",
            "type": "text",
            "value": undefined,
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Input
            data-__field={
              Object {
                "name": "password",
              }
            }
            data-__meta={
              Object {
                "name": "password",
                "originalProps": Object {
                  "disabled": false,
                  "placeholder": "Password",
                  "prefixCls": "ant-input",
                  "type": "password",
                },
                "ref": null,
                "rules": Array [
                  Object {
                    "message": "This field is required",
                    "required": true,
                  },
                ],
                "trigger": "onChange",
                "validate": Array [
                  Object {
                    "rules": Array [
                      Object {
                        "message": "This field is required",
                        "required": true,
                      },
                    ],
                    "trigger": Array [
                      "onChange",
                    ],
                  },
                ],
                "valuePropName": "value",
              }
            }
            disabled={false}
            id="password"
            onChange={[Function]}
            placeholder="Password"
            prefixCls="ant-input"
            type="password"
          />,
          "className": "isoInputWrapper",
          "colon": true,
          "hasFeedback": false,
          "prefixCls": "ant-form",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data-__field": Object {
              "name": "password",
            },
            "data-__meta": Object {
              "name": "password",
              "originalProps": Object {
                "disabled": false,
                "placeholder": "Password",
                "prefixCls": "ant-input",
                "type": "password",
              },
              "ref": null,
              "rules": Array [
                Object {
                  "message": "This field is required",
                  "required": true,
                },
              ],
              "trigger": "onChange",
              "validate": Array [
                Object {
                  "rules": Array [
                    Object {
                      "message": "This field is required",
                      "required": true,
                    },
                  ],
                  "trigger": Array [
                    "onChange",
                  ],
                },
              ],
              "valuePropName": "value",
            },
            "disabled": false,
            "id": "password",
            "onChange": [Function],
            "placeholder": "Password",
            "prefixCls": "ant-input",
            "type": "password",
            "value": undefined,
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Button
            block={false}
            ghost={false}
            htmlType="submit"
            loading={false}
            prefixCls="ant-btn"
            type="primary"
          >
            Sign In
          </Button>,
          "colon": true,
          "hasFeedback": false,
          "prefixCls": "ant-form",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": false,
            "children": "Sign In",
            "disabled": undefined,
            "ghost": false,
            "htmlType": "submit",
            "loading": false,
            "prefixCls": "ant-btn",
            "type": "primary",
          },
          "ref": null,
          "rendered": "Sign In",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <FormItem
            className="isoInputWrapper"
            colon={true}
            hasFeedback={false}
            prefixCls="ant-form"
          >
            <Input
              data-__field={
                Object {
                  "name": "email",
                }
              }
              data-__meta={
                Object {
                  "name": "email",
                  "originalProps": Object {
                    "disabled": false,
                    "placeholder": "Email",
                    "prefixCls": "ant-input",
                    "type": "text",
                  },
                  "ref": null,
                  "rules": Array [
                    Object {
                      "message": "Please enter a valid email address",
                      "pattern": /\\^\\[a-z0-9\\._%\\+-\\]\\+@\\[a-z0-9\\.-\\]\\+\\\\\\.\\[a-z\\]\\{2,\\}\\$/,
                      "required": true,
                    },
                  ],
                  "trigger": "onChange",
                  "validate": Array [
                    Object {
                      "rules": Array [
                        Object {
                          "message": "Please enter a valid email address",
                          "pattern": /\\^\\[a-z0-9\\._%\\+-\\]\\+@\\[a-z0-9\\.-\\]\\+\\\\\\.\\[a-z\\]\\{2,\\}\\$/,
                          "required": true,
                        },
                      ],
                      "trigger": Array [
                        "onChange",
                      ],
                    },
                  ],
                  "valuePropName": "value",
                }
              }
              disabled={false}
              id="email"
              onChange={[Function]}
              placeholder="Email"
              prefixCls="ant-input"
              type="text"
            />
          </FormItem>,
          <FormItem
            className="isoInputWrapper"
            colon={true}
            hasFeedback={false}
            prefixCls="ant-form"
          >
            <Input
              data-__field={
                Object {
                  "name": "password",
                }
              }
              data-__meta={
                Object {
                  "name": "password",
                  "originalProps": Object {
                    "disabled": false,
                    "placeholder": "Password",
                    "prefixCls": "ant-input",
                    "type": "password",
                  },
                  "ref": null,
                  "rules": Array [
                    Object {
                      "message": "This field is required",
                      "required": true,
                    },
                  ],
                  "trigger": "onChange",
                  "validate": Array [
                    Object {
                      "rules": Array [
                        Object {
                          "message": "This field is required",
                          "required": true,
                        },
                      ],
                      "trigger": Array [
                        "onChange",
                      ],
                    },
                  ],
                  "valuePropName": "value",
                }
              }
              disabled={false}
              id="password"
              onChange={[Function]}
              placeholder="Password"
              prefixCls="ant-input"
              type="password"
            />
          </FormItem>,
          <FormItem
            colon={true}
            hasFeedback={false}
            prefixCls="ant-form"
          >
            <Button
              block={false}
              ghost={false}
              htmlType="submit"
              loading={false}
              prefixCls="ant-btn"
              type="primary"
            >
              Sign In
            </Button>
          </FormItem>,
        ],
        "className": "isoSignInForm",
        "hideRequiredMark": false,
        "layout": "horizontal",
        "onSubmit": [Function],
        "prefixCls": "ant-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Input
              data-__field={
                Object {
                  "name": "email",
                }
              }
              data-__meta={
                Object {
                  "name": "email",
                  "originalProps": Object {
                    "disabled": false,
                    "placeholder": "Email",
                    "prefixCls": "ant-input",
                    "type": "text",
                  },
                  "ref": null,
                  "rules": Array [
                    Object {
                      "message": "Please enter a valid email address",
                      "pattern": /\\^\\[a-z0-9\\._%\\+-\\]\\+@\\[a-z0-9\\.-\\]\\+\\\\\\.\\[a-z\\]\\{2,\\}\\$/,
                      "required": true,
                    },
                  ],
                  "trigger": "onChange",
                  "validate": Array [
                    Object {
                      "rules": Array [
                        Object {
                          "message": "Please enter a valid email address",
                          "pattern": /\\^\\[a-z0-9\\._%\\+-\\]\\+@\\[a-z0-9\\.-\\]\\+\\\\\\.\\[a-z\\]\\{2,\\}\\$/,
                          "required": true,
                        },
                      ],
                      "trigger": Array [
                        "onChange",
                      ],
                    },
                  ],
                  "valuePropName": "value",
                }
              }
              disabled={false}
              id="email"
              onChange={[Function]}
              placeholder="Email"
              prefixCls="ant-input"
              type="text"
            />,
            "className": "isoInputWrapper",
            "colon": true,
            "hasFeedback": false,
            "prefixCls": "ant-form",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "data-__field": Object {
                "name": "email",
              },
              "data-__meta": Object {
                "name": "email",
                "originalProps": Object {
                  "disabled": false,
                  "placeholder": "Email",
                  "prefixCls": "ant-input",
                  "type": "text",
                },
                "ref": null,
                "rules": Array [
                  Object {
                    "message": "Please enter a valid email address",
                    "pattern": /\\^\\[a-z0-9\\._%\\+-\\]\\+@\\[a-z0-9\\.-\\]\\+\\\\\\.\\[a-z\\]\\{2,\\}\\$/,
                    "required": true,
                  },
                ],
                "trigger": "onChange",
                "validate": Array [
                  Object {
                    "rules": Array [
                      Object {
                        "message": "Please enter a valid email address",
                        "pattern": /\\^\\[a-z0-9\\._%\\+-\\]\\+@\\[a-z0-9\\.-\\]\\+\\\\\\.\\[a-z\\]\\{2,\\}\\$/,
                        "required": true,
                      },
                    ],
                    "trigger": Array [
                      "onChange",
                    ],
                  },
                ],
                "valuePropName": "value",
              },
              "disabled": false,
              "id": "email",
              "onChange": [Function],
              "placeholder": "Email",
              "prefixCls": "ant-input",
              "type": "text",
              "value": undefined,
            },
            "ref": [Function],
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Input
              data-__field={
                Object {
                  "name": "password",
                }
              }
              data-__meta={
                Object {
                  "name": "password",
                  "originalProps": Object {
                    "disabled": false,
                    "placeholder": "Password",
                    "prefixCls": "ant-input",
                    "type": "password",
                  },
                  "ref": null,
                  "rules": Array [
                    Object {
                      "message": "This field is required",
                      "required": true,
                    },
                  ],
                  "trigger": "onChange",
                  "validate": Array [
                    Object {
                      "rules": Array [
                        Object {
                          "message": "This field is required",
                          "required": true,
                        },
                      ],
                      "trigger": Array [
                        "onChange",
                      ],
                    },
                  ],
                  "valuePropName": "value",
                }
              }
              disabled={false}
              id="password"
              onChange={[Function]}
              placeholder="Password"
              prefixCls="ant-input"
              type="password"
            />,
            "className": "isoInputWrapper",
            "colon": true,
            "hasFeedback": false,
            "prefixCls": "ant-form",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "data-__field": Object {
                "name": "password",
              },
              "data-__meta": Object {
                "name": "password",
                "originalProps": Object {
                  "disabled": false,
                  "placeholder": "Password",
                  "prefixCls": "ant-input",
                  "type": "password",
                },
                "ref": null,
                "rules": Array [
                  Object {
                    "message": "This field is required",
                    "required": true,
                  },
                ],
                "trigger": "onChange",
                "validate": Array [
                  Object {
                    "rules": Array [
                      Object {
                        "message": "This field is required",
                        "required": true,
                      },
                    ],
                    "trigger": Array [
                      "onChange",
                    ],
                  },
                ],
                "valuePropName": "value",
              },
              "disabled": false,
              "id": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "prefixCls": "ant-input",
              "type": "password",
              "value": undefined,
            },
            "ref": [Function],
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Button
              block={false}
              ghost={false}
              htmlType="submit"
              loading={false}
              prefixCls="ant-btn"
              type="primary"
            >
              Sign In
            </Button>,
            "colon": true,
            "hasFeedback": false,
            "prefixCls": "ant-form",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "block": false,
              "children": "Sign In",
              "disabled": undefined,
              "ghost": false,
              "htmlType": "submit",
              "loading": false,
              "prefixCls": "ant-btn",
              "type": "primary",
            },
            "ref": null,
            "rendered": "Sign In",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
